news/feature-3050.md

```diff
@@ -0,0 +1 @@
+`rewrite`: Added type hinting support.
```

Comment:
I would say the PR introduces the possibility to use template options, but I don't think there is any type-hint for a template. I would also say that this is a bugfix: I think template options should be usable everywhere we a template is accepted.

About type-hint:
I've checked the code, I see that there is some intention to support this, but on LogTemplate side there is nothing about typehints, except that it's an member of LogTemplate.

Reply:
@Kokan I might have misunderstood this, can you double-check please? 10Q!

Reply:
Not it was my mistake, only concluded that from the grammar intent. But as @gaborznagy explained the user must act upon, which template does not do.

Reply:
So if I understand right, this is just a refactor? I removed the news entry, but I am happy to put it back, if needed.

Reply:
Yes, just refactor.